### PR TITLE
Major rewrite. A thin layer on top of mds in production.

### PR DESCRIPTION
**Pull request by arkilic**

### CODE REVIEW

File: metadataservice/schema/run_start.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
             "description": "Name of project that this run is part of"
         },
         "sample": {
-            "type": ["object"],
+            "type": ["string"],
```
Comment (danielballan) [inline metadataservice/schema/run_start.json]: In the [corresponding line the in bluesky PR](https://github.com/NSLS-II/bluesky/pull/375/files#diff-7f868e2356e2827e8024a1cba664c267R8), I allowed it to be a string or an objec*. Changing it to string would require a migration, and I don't want to do that anyway. (P.S. I know you squeezed this work in amidst school stuff -- no worries.)

**Comment by danielballan** [comment]
Looks like this needs a rebase.

**Comment by arkilic** [comment]
Ugh, you beat me to it, I was pushing it :)

**Comment by danielballan** [comment]
Haha, cool. This still needs a rebase.

**Comment by arkilic** [comment]
Yes, before I do that, there are a few other things I want to discuss with you and Tom. Let's do this tmrw a.m.?

**Comment by danielballan** [comment]
Fine by me. No rush on this.

**Comment by arkilic** [comment]
I would like this deployed and tested on a beamline as soon as we can. Amostra and analysisstore is also coming along so the sooner this happens, the more attention we can pay to the new stuff

**Comment by tacaswell** [comment]
this needs a rebase.

**Comment by arkilic** [comment]
@hhslepicka this is the testing mechanism we have been talking about, [I put it together](http://nerdist.com/wp-content/uploads/2015/12/Speedy-Gonzales.jpg). 
mdservice tests start the mdservice server on the fly (using a random db), perform the tests, and kill the process in mds_setup() and mds_teardown().

**Comment by tacaswell** [comment]
how did the amount of code go _up_?

**Comment by arkilic** [comment]
This version has tests, the old version did not. All that _version.py etc is a lot of code

**Comment by arkilic** [comment]
Still WIP, it is not ready for review yet.

**Comment by tacaswell** [comment]
Turned travis on.

**Comment by arkilic** [comment]
@hhslepicka could explain @tacaswell  why we need [these](https://github.com/NSLS-II/metadataservice/pull/5/commits/f1fa8d7c6d8e95d0113894e8e614cdc645507631#diff-1f28f732bdc45c1260c415c6bf8a14e2R21)?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  - 3.5
+
+before_install:
+  - git clone https://github.com/nsls-ii/nsls2-ci ~/scripts
+  - . ~/scripts/install-miniconda.sh ~/
+
+install:
+  - conda create -n testenv --yes nose python=$TRAVIS_PYTHON_VERSION pymongo ujson numpy tornado jsonschema requests pyyaml pytz pytest doct -c lightsource2
+  - source activate testenv
+  - 'pip install https://github.com/NSLS-II/metadatastore/zipball/master#egg=metadatastore'
```
Comment (klauer) [inline .travis.yml]: What's with the quotes?

File: metadataservice/server/engine.py [hunk 3]
```diff
@@ -61,134 +25,129 @@ def set_default_headers(self):
...
-        """Abstract method, here to show it exists explicitly. Useful for streaming client"""
+        """Abstract method, here to show it exists explicitly.
+        Useful for streaming client"""
         pass
 
+    def queryable(self, func):
+        try:
+            return self.queryables[func]
+        except KeyError as err:
+            utils.report_error(500, 'Provided query method {} is not supported'.format(func))
```
Comment (klauer) [inline metadataservice/server/engine.py]: 👍 for sensible error messages

**Comment by klauer** [comment]
👍

**Comment by hhslepicka** [comment]
@tacaswell what is your concern regarding the open access control header parameters?

**Comment by tacaswell** [comment]
Smells like the tornado equivalent to `chmod a+rw -R *`

File: .travis.yml (hunk 2 above)
Comment (arkilic) [inline .travis.yml]: I have no idea. I saw it somewhere else in our code base and added it here to get travis to work. Any idea @ericdill?
PS: I need some feature we recently added to mds master for mdservice to work (i.e. no error on import w/o config)

**Comment by hhslepicka** [comment]
We need it because most of the time, if not always, the clients will be in a machine other than where the service is running. If the client follows the Same-origin policy the request will be denied if the flag is not there. Open to suggestions here! =)

File: .travis.yml (hunk 2 above)
Comment (ericdill) [inline .travis.yml]: The quotes are not needed.  It may have been needed at some point to make sure travis understood how to execute this command

Comment (arkilic) [inline .travis.yml]: 🚲 🏠  complete @klauer

**Pull request merged**